code/modules/admin/verbs/debug.dm

```diff
@@ -234,16 +234,44 @@ But you can call procs that are of type /mob/living/carbon/human/proc/ for that
 	if(M.ckey)
 		if(alert("This mob is being controlled by [M.key]. Are you sure you wish to assume control of it? [M.key] will be made a ghost.",,"Yes","No") != "Yes")
 			return
-		else
-			M.ghostize(FALSE)
+	if(!M || QDELETED(M))
+		to_chat(usr, "<span class='warning'>The target mob no longer exists.</span>")
+		return
+	var/mob/adminmob = mob
+	M.ghostize(FALSE)
+	M.ckey = ckey
+	if(isobserver(adminmob))
+		qdel(adminmob)
 	message_admins("<span class='adminnotice'>[key_name_admin(usr)] assumed direct control of [M].</span>")
```

Comment:
This is post ckey transfer, usr is the keyless mob now.

Reply:
Thanks